packages/opencensus-web-instrumentation-zone/src/interaction-tracker.ts

```diff
@@ -33,7 +33,12 @@ export type AsyncTask = Task & {
 };
 
 export class InteractionTracker {
+  // Allows to track several events triggered by the same user interaction in the right Zone.
+  private shouldCreateNewZone = true;
+  private currentZone: Zone = Zone.root;
+
   constructor() {
+    const interactionTracker: InteractionTracker = this;
```

Comment:
Why do we need to capture `this` in a variable?

Reply:
This is necessary for monkey-patched methods (e.g. runTask) as the scope of these change we need to keep track of the interaction tracker.

Reply:
Can you try using arrow functions `() => ` rather than a `function` declaration? 

Arrow functions capture the `this` variable automatically, see https://hackernoon.com/javascript-es6-arrow-functions-and-lexical-this-f2a3e2a5e8c4